js-core/homeworks/homework-3/src/main.js

```diff
@@ -0,0 +1,153 @@
+        }
+        return modifyObj
+    }
+    let someObjMod = returnObj(myObj)
+    console.log(someObjMod)
+
+
+//===================================================================
+    /*
+    1. Напишите функцию которая принимает в
+    качестве аргумента массив и элемент массива
+    и добавляет элемент в конец массива
+    */
+
+    let arr = [];
+    arr[0] = 'privet', arr[1] = 'poka', arr[2] = 'zdrastvuite'; //default values
+
+    function addToArray(arrParam,elemParam){
+        return arrParam[arrParam.length] = elemParam;
+    }
+
+    addToArray(arr,'aloe') //add aloe
+    addToArray(arr,'vera') //add vera
+    addToArray(arr,'how are you') //add how are you
+
+    console.log(arr)
+//===================================================================
+    /*
+    Напишите функцию которая получает 3 параметра
+    и возвращает объект типа:
+        {
+        argument1: param1,
+        argument2: param2,
+        argument3: param3,
+        }
+    */
+    let userNames = [{ name: 'Egor' }, { name: 'Katya' }, { name: 'Vera' }];
+
+    function simpleObjGenerator (param1,param2,param3){
+        let obj = {
+            argument1: param1,
+            argument2: param2,
+            argument3: param3
+        }
+        return obj
+        
+    }
+    let resultSimple = simpleObjGenerator('protocol', { url: '22' }, 8000)
+    let resultParams = simpleObjGenerator(77,userNames,'privet kak dela chto novogo'.toUpperCase())
+    console.log(resultSimple)
+    console.log(resultParams)
+
+//===================================================================
+    /*
+    Напишите функцию которая принимает 3 аргумента,
+    третий аргумент - это объект.
+    Функция создает новый объект и добавляет ключ
+    это первый аргумент, а значение - второй аргумент
+    Проверяет если есть свойство name в переданном объекте,
+    тогда добавляет данное свойство и возвращает новый объект
+    */
+
+    var myName = { name: 'Oleg' };
+
+    function addNameToUser(newKey, newValue, userName) {
+        let objUser = {}
+        if(userName.name){
+            objUser.name = userName.name
+        }   
+        objUser[newKey] = newValue        
+        return objUser
+    }
+    let objResult = addNameToUser('mood','happy',myName);
+    let addResult = addNameToUser('surname','pupkin',{})
+    let addResult2 = addNameToUser('name','vasya',{})
+
+    console.log(objResult)
+    console.log(addResult)
+    console.log(addResult2)
+
+//===================================================================
+
+    /*
+    -> @@ SUPER !
+    Напишите функцию, которая будет
+    возвращать 'Fizz' если передаваемый параметр кратен 3,
+    'Buzz', если передаваемый параметр кратен 5,
+    'FizzBuzz' - если параметер кратен 3 и 5
+    Если передаваемое число не кратно 3 или 5, то вернуть указанный параметр
+    */
+
+    function fizzBuzz(param){
+        if((param % 3) == 0 && (param % 5) == 0){return 'FizzBuzz'}
```

Comment:
There some styles changes required, please don't make a single line with if statement.

Furthermore, some conditionals can be simplified, let we think can we improve it?